### PR TITLE
[compiler][patch] Fix jsx non-ascii regex pattern

### PR DESCRIPTION
**Pull request by mofeiZ**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #30389
* #30386
* __->__ #30382

### CODE REVIEW

File: compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/fbt/error.todo-fbt-param-with-unicode.expect.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  params: [{ name: "Jason" }],
+};
+
+```
+
+
+## Error
+
+```
+Property arguments[0] of CallExpression expected node to be of a type ["Expression","SpreadElement","JSXNamespacedName","ArgumentPlaceholder"] but instead got "JSXExpressionContainer"
```
Comment (mofeiZ) [inline compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/fbt/error.todo-fbt-param-with-unicode.expect.md:27]: This error is not caught and handled by Forget as it occurs in `babel-plugin-fbt`. Going to go back to hack in a check in `MemoizeFbtAndMacroOperandsInSameScope` for now, but ideally we would have some type / annotation on `Identifier` instances to know which strings are fbt operands

File: compiler/packages/babel-plugin-react-compiler/src/ReactiveScopes/CodegenReactiveFunction.ts [hunk 2]
```diff
@@ -2206,7 +2206,7 @@ function codegenInstructionValue(
  * https://en.wikipedia.org/wiki/List_of_Unicode_characters#Control_codes
  */
 const STRING_REQUIRES_EXPR_CONTAINER_PATTERN =
-  /[\u{0000}-\u{001F}|\u{007F}|\u{0080}-\u{FFFF}]|"/u;
+  /[\u{0000}-\u{001F}\u{007F}\u{0080}-\u{FFFF}]|"/u;
```
Comment (mofeiZ) [inline compiler/packages/babel-plugin-react-compiler/src/ReactiveScopes/CodegenReactiveFunction.ts:2209]: Previously, we were matching `|` literals

Comment (poteto) [reply compiler/packages/babel-plugin-react-compiler/src/ReactiveScopes/CodegenReactiveFunction.ts:2209]: I can't recall why we needed to escape the `u{...`, does it work without them?

**Review by poteto — APPROVED** [review]

File: compiler/packages/babel-plugin-react-compiler/src/ReactiveScopes/CodegenReactiveFunction.ts (hunk 2 above)
Comment (mofeiZ) [reply compiler/packages/babel-plugin-react-compiler/src/ReactiveScopes/CodegenReactiveFunction.ts:2209]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/unicode#description

**Pull request merged**